packages/api/src/apps/api/app.ts

```diff
@@ -120,7 +130,7 @@ export default async (env: ProcessEnv) => {
     // these services can optionally be configured with a config object, but currently they are undefined or have defaults
     blocks: Services.Blocks(undefined, appState),
     emps: Services.EmpState({ debug }, appState),
-    registry: await Services.Registry({ debug }, appState),
+    registry: await Services.Registry({ debug }, appState, (data) => serviceEvents.emit("empRegistry", data)),
```

Comment:
some context here. this pr splits up contract discovery from contract state fetching, which means a contract could be discovered at some point, but then not have state until state is fetched during its separate update loop.  This adds a pattern to services where they can emit any kind of event to trigger external processing. In this case when a new contract is discovered, it emits an event, which gets tagged and passed to a global event emitter `serviceEvents`. This is listened to further down to trigger a singular update on the new contract.